src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -3400,7 +3400,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     contribute to incremental build inconsistencies.
     ============================================================
     -->
-  <Target Name="_GenerateCompileDependencyCache" Condition="'$(DesignTimeBuild)' != 'true' and '$(BuildingProject)' != 'true'" DependsOnTargets="ResolveAssemblyReferences">
+  <Target Name="_GenerateCompileDependencyCache" Condition="'$(DesignTimeBuild)' != 'true' and '$(BuildingProject)' == 'true'" DependsOnTargets="ResolveAssemblyReferences">
```

Comment:
Pedantically checking ourselves here:

The [documented check for "is a design time build"](https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md#determining-whether-a-target-is-running-in-a-design-time-build) is 

```
'$(DesignTimeBuild)' == 'true' OR '$(BuildingProject)' != 'true'
```

We want its inverse here, so

```
!( '$(DesignTimeBuild)' == 'true' OR '$(BuildingProject)' != 'true' )
```

We can use [DeMorgan's theorems](https://www.allaboutcircuits.com/textbook/digital/chpt-7/demorgans-theorems/):

```
!('$(DesignTimeBuild)' == 'true') AND !('$(BuildingProject)' != 'true')

'$(DesignTimeBuild)' != 'true') AND '$(BuildingProject)' == 'true'
```

Which is what this PR does. So 👍.